src/main/java/frc/robot/subsystems/localization/Localizer.java

```diff
@@ -0,0 +1,193 @@
+
+
+PhotonCamera camera = new PhotonCamera("Slotheye");
+
+
+
+    visionThread =
+        new Thread(
+            () -> {
+              // Get the UsbCamera from CameraServer
+
+              SmartDashboard.putBoolean("Camera ON", true);
+              SmartDashboard.putNumber("camera Count", count++);
+              // Set the resolution
+
+              // // Get a CvSink. This will capture Mats from the camera
+              // CvSink cvSink = CameraServer.getVideo();
+
+              // // Setup a CvSource. This will send images back to the Dashboard
+              // CvSource outputStream = CameraServer.putVideo("Rectangle", 640, 480);
+
+              // // Mats are very memory expensive. Lets reuse this Mat.
+              // Mat mat = new Mat();
+              // Mat grayMat = new Mat();
+              // ArrayList<Integer> tags = new ArrayList<>();
+
+
+              // This cannot be 'true'. The program will never exit if it is. This
+              // lets the robot stop this thread when restarting robot code or
+              // deploying.
+              while (!Thread.interrupted()) {
+
+                // // Tell the CvSink to grab a frame from the camera and put it
+                // // in the source mat.  If there is an error notify the output.
+                // if (cvSink.grabFrame(mat) == 0) {
+                //   // Send the output the error.
+                //   outputStream.notifyError(cvSink.getError());
+                //   // skip the rest of the current iteration
+                //   continue;
+                // }
+
+                // Imgproc.cvtColor(mat, grayMat, Imgproc.COLOR_RGB2GRAY);
+                // detections = detector.detect(grayMat);
+
+                PhotonPipelineResult result = camera.getLatestResult();
+                PhotonTrackedTarget target = result.getBestTarget();
+
+                if(target!=null)
+                {
+
+                  SmartDashboard.putNumber("tag seen", target.getFiducialId());
+
+                  int tagID = target.getFiducialId();
+                  Transform3d transformPose = target.getBestCameraToTarget();
+                  Pose3d plusPose = findPoseTransform(transformPose, tagID);
+                  SmartDashboard.putString("tag", "" + tagID);
+                  SmartDashboard.putString("transform", transformPose.toString());
+                  SmartDashboard.putString("plus", plusPose.toString());                
+              }
+
+                // Put a rectangle on the image
+                // Imgproc.rectangle(
+                //     mat, new Point(100, 100), new Point(400, 400), new Scalar(255, 255, 255), 5);
+
+                // Give the output stream a new image to display
+                // outputStream.putFrame(mat);
+              }
+            });
+
+  }
+
+  public void start() {
+    visionThread.setDaemon(true);
+    visionThread.start();
+  }
+
+  private Pose3d findPoseTransform(Transform3d pose, int tagID){ 
```

Comment:
It looks like this function will return an empty `Pose3d` if no tag is found. Since this is a placeholder value and should be treated differently by whatever is calling this function, I would like to see a comment letting the programmer know.

Reply:
You're also using `Optional<Pose3d>` here- you could just return that. That way, the user knows explicitly whether to check whether the transform was found.